test/typescript/typescript.test.ts

```diff
@@ -295,6 +311,65 @@ describe("jestConfig", () => {
     expect(jest.globals["ts-jest"].tsconfig).toStrictEqual("bar");
     expect(jest.globals["ts-jest"].shouldBePreserved).toStrictEqual(true);
   });
+
+  test("uses default values and jestVersion < 29", () => {
+    const prj = new TypeScriptProject({
+      defaultReleaseBranch: "main",
+      name: "test",
+      jestOptions: {
+        jestVersion: "^27",
+        jestConfig: {
+          globals: {
+            "ts-jest": {
+              shouldBePreserved: true,
+              tsconfig: "tsconfig.dev.json",
+            },
+          },
+        },
+      },
+    });
+    const snapshot = synthSnapshot(prj);
+    const jest = snapshot["package.json"].jest;
+    expect(jest.preset).toStrictEqual("ts-jest");
+    expect(jest.globals["ts-jest"].tsconfig).toStrictEqual("tsconfig.dev.json");
+    expect(jest.globals["ts-jest"].shouldBePreserved).toStrictEqual(true);
+  });
+
+  test("uses default values and jestVersion >= 29", () => {
+    const expectedTransformConfig = {
+      "^.+\\.(ts|tsx)$": [
+        "ts-jest",
+        {
+          tsconfig: "tsconfig.dev.json",
+        },
+      ],
+    };
+
+    const prj = new TypeScriptProject({
+      defaultReleaseBranch: "main",
+      name: "test",
+      jestOptions: {
+        jestVersion: "^29",
+        jestConfig: {
+          globals: {
+            "ts-jest": {
+              shouldBePreserved: true,
+            },
+          },
+          transform: {
+            "^.+\\.(ts|tsx)$": new Transform("ts-jest", {
+              tsconfig: "tsconfig.dev.json",
+            }),
+          },
+        },
+      },
+    });
+    const snapshot = synthSnapshot(prj);
+    const jest = snapshot["package.json"].jest;
+    expect(jest.preset).toStrictEqual("ts-jest");
+    expect(jest.globals["ts-jest"].shouldBePreserved).toStrictEqual(true);
+    expect(jest.transform).toStrictEqual(expectedTransformConfig);
+  });
```

Comment:
You are explicitly setting the config and then testing it's there. That's not checking our new code at all!
```suggestion
  test("uses transforms for ts-jest >= 29", () => {
    const prj = new TypeScriptProject({
      defaultReleaseBranch: "main",
      name: "test",
      jestOptions: {
        jestVersion: "^29",
      },
    });
    
    const expectedTransformConfig = {
      "^.+\\.(ts|tsx)$": [
        "ts-jest",
        {
          tsconfig: "tsconfig.dev.json",
        },
      ],
    };
    
    const snapshot = synthSnapshot(prj);
    const jest = snapshot["package.json"].jest;
    expect(jest.preset).toStrictEqual("ts-jest");
    expect(jest.transform).toStrictEqual(expectedTransformConfig);
  });
```

Same applied to the other tests